elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/dynamic/compile/ElideDynamicInMemoryClassLoader.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.spring.dynamic.compile;
+
+import com.google.common.collect.Sets;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Set;
+
+/**
+ * ClassLoader for dynamic configuration.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class ElideDynamicInMemoryClassLoader extends ClassLoader {
+
+    private Set<String> classNames = Sets.newHashSet();
+
+    public ElideDynamicInMemoryClassLoader(ClassLoader parent, Set<String> classNames) {
+        super(parent);
+        setClassNames(classNames);
+    }
+
+    public void setClassNames(Set<String> classNames) {
```

Comment:
What is the purpose of this public setter method that overrides the constructor initialization?  Can we remove it?